### ISSUE TITLE
Flyout menus have incorrect focus outline

### ISSUE THREAD

**Issue by MichaHuhn** [tailwindlabs/tailwind-plus-issues#1711]
**What component**
- URL for category: https://tailwindcss.com/plus/ui-blocks/marketing/elements/flyout-menus
- Affected components: All **Vue** components with a flyout menu (including header components).

**Describe the bug**
When clicking outside the flyout menu to close it, the focus outline flashes and stays active.

**To Reproduce**
The flyout menu component uses the Vue popover component from Headless UI.
The issue somehow doesn't happen on this page: https://tailwindcss.com/plus/ui-blocks/marketing/elements/flyout-menus
I don't know why.
However, it happens on the Headless UI page: https://tailwindcss.com/plus/ui-blocks/marketing/elements/flyout-menus
It also happens when using the flyout menu component in a Vue project. That's why I submitted this GitHub issue.

1. Go to https://headlessui.com/v1/vue/popover
2. Open and close the popover component a few times.

Sometimes the issue doesn't happen for whatever reason.
In this case try to reload the page and try again.

**Demonstration**

https://github.com/user-attachments/assets/27e8ca0c-b117-4534-a1c0-a47ca5b577c4

**Expected behavior**
The focus outline should behave like `:focus-visible`, so it should only be active when navigating with the keyboard but not with the mouse.

**Browser/Device**
 - Browser: Chrome
 - OS: Windows 10

**Additional context**
Maybe the issue is related to this: https://github.com/tailwindlabs/headlessui/issues/1694

**Comment by thecrypticace** [comment]
Hey! So this is a known issue in Headless UI. We released [Elements](https://tailwindcss.com/blog/vanilla-js-support-for-tailwind-plus) on Friday and it **does not have this problem** and works pretty well with Vue if you want to give that a try.

Unfortunately, fixing this in Headless UI is a bit more complicated so it's not likely to change any time soon (sorry!) as it has to do with issues around using browser APIs to programmatically focus elements, how events are handled, if a clicked element was focusable, etc… However, [there is a workaround](https://github.com/tailwindlabs/headlessui/issues/2862#issuecomment-1840891819) — Headless UI adds a `data-headlessui-focus-visible` attribute to the `html` element when using the keyboard which you can use in conjunction with CSS to control the styling.

You can use a class like `not-in-data-headlessui-focus-visible:focus:outline-none` to remove the outline on an element when the keyboard was not used.

For example, something like this would work to remove the outline that would otherwise be there:
```html
<script setup>
  import { Popover, PopoverButton, PopoverPanel } from '@headlessui/vue'
</script>
<template>
  <Popover class="relative">
    <PopoverButton
      class="
        px-3 py-2 text-white focus-visible:outline focus-visible:outline-white/75
        not-in-data-headlessui-focus-visible:focus:outline-none
      "
    >
      <span>Solutions</span>
    </PopoverButton>
    <PopoverPanel class="absolute z-10">
      <div class="grid grid-cols-2">
        <a href="/analytics">Analytics</a>
        <a href="/engagement">Engagement</a>
        <a href="/security">Security</a>
        <a href="/integrations">Integrations</a>
      </div>
    </PopoverPanel>
  </Popover>
</template>
```

Hopefully that helps 👍

**Comment by MichaHuhn** [comment]
`not-in-data-headlessui-focus-visible:focus:outline-none` works.
Thank you very much!
